[qinboan] iP

qinboan opened the pull request:
### DiveBot
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DiveBot frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

qinboan committed:
Level-0

qinboan committed:
Level-2

qinboan committed:
Level-1

qinboan committed:
Merge branch 'master' of https://github.com/qinboan/ip

qinboan committed:
Level-2

qinboan committed:
Level-3

qinboan committed:
Level-4

qinboan committed:
A-TextUiTesting

qinboan committed:
Level-5

qinboan committed:
Level-6

qinboan committed:
Level-7

qinboan committed:
Level-8

qinboan committed:
Merge branch 'branch-Level-8'

JerryO3 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,31 @@
+import java.time.LocalDate;
```
You may want to package your classes so that contributors are better able to understand the code base!

JerryO3 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,31 @@
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
```
Java Documentation here will be good to explain the functionality of the class.

JerryO3 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,31 @@
+import java.time.format.DateTimeParseException;
+public class Deadline extends Task {
+    private LocalDate byDate;
+
```
Also a good idea to provide Java Documentation for constructors, especially since you have exception handling logic that makes this method non-trivial 😄

JerryO3 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,241 @@
+            saveTasksToFile();
+        } else {
+            throw new DukeException("Task does not exist.");
+            //System.out.println("Task does not exist.");
```
You can consider deleting this commented code segment, and instead use version control functionality to store this code as a previous commit/other branch. Would make your code neater too!

JerryO3 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,241 @@
+import java.util.Scanner;
```
As mentioned in another comment, would be good if you can provide Java Documentation for all your non-trivial methods and class so that other developers are able to better understand your codebase.

JerryO3 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,42 @@
+import java.time.LocalDate;
```
Same comments about Javadocs headers

JerryO3 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,26 @@
+public class Task {
```
Missing Javadocs headers, mentioned in another comment.

kervyntan replied:
Missing Javadocs headers

JerryO3 reviewed (commented):
Would be good if you could provide Java Documentation for your classes and methods for others' reference! Also, each class should ideally be packaged for easier handling.

qinboan committed:
A-MoreOOP

qinboan committed:
A-Packages

qinboan committed:
Merge branch 'master' of https://github.com/qinboan/ip into add-gradle-support

qinboan committed:
changed buildgradle

qinboan committed:
A-JUnit

qinboan committed:
A-Jar

kervyntan commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,25 @@
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "[X]" : "[ ]");
```
Parantheses not needed around this expression!

qinboan committed:
A-JavaDoc

qinboan committed:
Merge branch '!branch-A-JavaDoc'

qinboan committed:
A-CodingStandard

qinboan committed:
Merge branch 'branch-A-CodingStandard'

qinboan committed:
Level-9

qinboan committed:
Level-9

kervyntan reviewed (commented):
Overall good job! Just generally small things to take note of

qinboan committed:
Level-10

qinboan committed:
Improve code quality by refactoring Parser class

Enhance the maintainability of the Parser class by adding new methods that encapsulate the logic for command processing. This refactoring helps in segregating responsibilities within the class, making the codebase more modular and easier to understand.

- Extract command-specific logic into separate methods
- Simplify the main parse method to improve readability
- Ensure each new method has a single responsibility

This change not only improves code quality but also sets a foundation for easier future extensions and potential addition of new commands.

qinboan committed:
A-Assertions

qinboan committed:
Merge branch 'branch-A-Assertions'

qinboan committed:
A-CodeQuality

qinboan committed:
A-CodeQuality

qinboan committed:
Merge branch 'branch-A-CodeQuality'

qinboan committed:
Add Fixed duration tasks

qinboan committed:
Update README

qinboan committed:
Update ReadMe

qinboan committed:
Merge pull request #2 from qinboan/branch-ReadMe

Update ReadMe

qinboan committed:
Update README.md

qinboan committed:
Update README.md

qinboan committed:
Update Build.gradle
